### ISSUE TITLE
[XrdHttp] Incorrect response code to HTTP PUT

### ISSUE THREAD

**Issue by bbockelm** [xrootd/xrootd#2382]
When developing https://github.com/xrootd/xrootd/pull/2381, I noticed that, when uploading a new object, we unconditionally respond with a HTTP 200.  This is incorrect behavior.

As outlined in https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT, there are two typical responses:
- If no object existed beforehand, `201 Created` must be used.
- If the object was preexisting, `200 OK` or `204 No Content` is acceptable (the later case is when there's no message body in the response).

We should open first with `kXR_new` and, if that fails because the object already exists, retry with `kXR_delete`.

**Comment by ccaffy** [comment]
Thanks @bbockelm , I see you already fixed one part which is return 201 instead of 200: https://github.com/xrootd/xrootd/pull/2381/commits/aa60af1f7fe12b2210e8aadf17020e1c137859f1

**Comment by bbockelm** [comment]
Yes -- sorry, I forgot to link it in.  I think https://github.com/xrootd/xrootd/commit/aa60af1f7fe12b2210e8aadf17020e1c137859f1 provides a fix for the most common case (as my gut feeling is "creating new object" is indeed the most common case!).  However, when I realized that a full fix requires deeper changes -- e.g., potentially opening the file twice -- I separated the complete solution into this issue.

**Comment by amadio** [comment]
Fixed in #2381, which was merged for 5.7.3.
